test_runner/regress/test_tenant_size.py

```diff
@@ -436,9 +450,16 @@ def test_single_branch_get_tenant_size_grows(
 
             size = http_client.tenant_size(tenant_id)
             prev = collected_responses[-1][1]
-            assert (
-                size > prev
-            ), "even though rows have been deleted, the tenant_size should increase"
+
+            if current_lsn - initdb_lsn > gc_horizon:
+                assert (
+                    size >= prev
+                ), "tenant_size may grow or not grow, because we only add gc_horizon amount of WAL to initial snapshot size"
+            else:
+                assert (
+                    size > prev
+                ), "tenant_size should grow, because we continue to add WAL to initial snapshot size"
```

Comment:
Assuming this is a correct property which must hold, maybe refactor it to an test-function internal function which allows these to be replaced as:

```python
check_growth(current_lsn, prev, size)
```

`check_growth` is defined somewhere after `initdb_lsn` and `gc_horizon` have been defined:

```
def check_growth(current_lsn, prev_size, size_now):
  if current_lsn - initdb_lsn > gc_horizon:
    assert size_now >= prev_size, "long rationale"
  else:
    assert size_now > prev_size, "long rationale"
```